iceoryx_examples/automotive_soa/include/owl/kom/event_publisher_uds.inl

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_PUBLISHER_UDS_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_PUBLISHER_UDS_INL
+
+#include "owl/kom/event_publisher_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventPublisher<T, EventTransmission::UDS>::EventPublisher(const core::String& service,
+                                                                 const core::String& instance,
+                                                                 const core::String& event) noexcept
+    : m_publisher({service, instance, event}, {HISTORY_CAPACITY, iox::NodeName_t(), OFFERED_ON_CREATE})
+    , m_instanceId(instance)
+{
+}
+
+template <typename T>
+inline std::unique_ptr<T> EventPublisher<T, EventTransmission::UDS>::Allocate()
+{
+    // The proxy needs some time to discover the service and create the EventSubscriber with the UDS server,
+    // hence the creation of the UDS client is done here
+    if (!m_uds.isInitialized())
+    {
+        m_uds = iox::posix::UnixDomainSocket::create(m_instanceId, iox::posix::IpcChannelSide::CLIENT)
+                    .expect("Failed to create UNIX domain socket!");
+    }
+
+    // Allocate the memory on the heap
+    return std::make_unique<SampleType>();
+}
+
+template <typename T>
+void EventPublisher<T, EventTransmission::UDS>::Send(std::unique_ptr<SampleType> userSamplePtr)
+{
+    if (!userSamplePtr)
+    {
+        std::cerr << "Provided empty sample pointer!" << std::endl;
```

Comment:
Since `Send` can fail in various cases I would return a `bool` to indicate success. Otherwise we have somehow an API which can fail but the developer will never know.

Reply:
Changed.